support/widgetConfigUtils/basemapToggle.ts

```diff
@@ -0,0 +1,110 @@
+import BasemapToggle from "esri/widgets/BasemapToggle";
+import Basemap from "esri/Basemap";
+
+//////////////////////////////
+// Public Module Functions
+//////////////////////////////
+
+/** The original basemap for the Webmap */
+let _originalBasemap: __esri.Basemap;
+/** The alternate basemap in the BasemapToggle */
+let _nextBasemap: __esri.Basemap;
+
+/**
+ * Adds BasemapToggle to Application, including logic to make 
+ * integrations with the Config Panel function properly
+ * @param props 
+ */
+export async function addBasemapToggle(props: esriWidgetProps): Promise<__esri.BasemapToggle> {
+
+    // Variables
+    const { view, config, propertyName, portal } = props;
+    const { basemapToggle, basemapTogglePosition, basemapSelector, nextBasemap } = config;
+    /** Depending on which is defined, use "basemapSelector"
+     * (which is an id, ex: db6a43e20e09486b9444edcd77e0f906) or "nextBasemap" 
+     * (which is a well-known-basemap-id: https://developers.arcgis.com/javascript/latest/api-reference/esri-Map.html#basemap).
+     * The "nextBasemap" config variable is kept for backward compatibilty with the Config Panel,
+     * whereas "basemapSelector" is the new config variable */
+    const alternateBasemapId: string = basemapSelector || nextBasemap;
+    let basemapToggleInstance: __esri.BasemapToggle;
+    
+    const node = _findNode("esri-basemap-toggle");
+    if (node) basemapToggleInstance = view.ui.find("basemapToggle") as __esri.BasemapToggle;
+
+    // Save original basemap
+    if(!_originalBasemap) _originalBasemap = view.map.basemap;
+
+    // Remove BasemapToggle
+    if (!basemapToggle) {
+        if (node) {
+            _resetBasemapsInToggle(basemapToggleInstance, _originalBasemap);
+            view.ui.remove(node);
+            basemapToggleInstance.destroy();
+        }
+        return null; // BasemapToggle has been removed
+    }
+
+    // setup nextBasemap
+    _nextBasemap = await _getBasemap(alternateBasemapId, portal);
+
+    if (node && propertyName === "basemapSelector") {
+        _resetBasemapsInToggle(basemapToggleInstance, _originalBasemap, _nextBasemap);
+    } else if (node && propertyName === "basemapTogglePosition") {
+        view.ui.move(node, basemapTogglePosition);
+    } else if (propertyName === "basemapToggle") {
+        basemapToggleInstance = new BasemapToggle({
+            view,
+            nextBasemap: _nextBasemap,
+            id: "basemapToggle"
+        });
+
+        view.ui.add(
+            basemapToggleInstance,    
+            basemapTogglePosition
+        );
+    }
+    return basemapToggleInstance;
+}
+
+
+//////////////////////////////
+// Private Module Functions
+//////////////////////////////
+
+/** 
+ * Creates Basemap instance properly from either a well-known-basemap-id, or
+ * from a webmap id
+ */
+async function _getBasemap(id: string, portal?: __esri.Portal): Promise<__esri.Basemap> {
+    let basemap: __esri.Basemap = Basemap.fromId(id);
```

Comment:
I think this could be simplified as...

```
return Basemap.fromId(id) || new Basemap({
            portalItem: {
                id,
                portal
            }
        }).loadAll();
```

Reply:
Replaced with this